timemachine/ff/handlers/utils.py

```diff
@@ -32,6 +39,68 @@ def canonicalize_bond(arr):
         return arr
 
 
+@wraps(canonicalize_bonded_ixn)
+def canonicalize_bond(bond):
+    assert len(bond) == 2
+    return canonicalize_bonded_ixn(bond)
+
+
+@wraps(canonicalize_bonded_ixn)
+def canonicalize_angle(angle):
+    assert len(angle) == 3
+    return canonicalize_bonded_ixn(angle)
+
+
+@wraps(canonicalize_bonded_ixn)
+def canonicalize_proper_torsion(torsion):
+    assert len(torsion) == 4
+    return canonicalize_bonded_ixn(torsion)
+
+
+@wraps(canonicalize_bonded_ixn)
+def canonicalize_improper_torsion(torsion):
+    """
+    WIP: extract from two definitions:
+    1. https://github.com/proteneer/timemachine/blob/8d6bd25a143aa81e8b3b8c6a33e6e03afe272c56/timemachine/ff/handlers/bonded.py#L226-L229
+    2. https://github.com/proteneer/timemachine/blob/8d6bd25a143aa81e8b3b8c6a33e6e03afe272c56/timemachine/ff/handlers/bonded.py#L206-L213
+    """
+
+    # approach 1: sort neighbors (used for FF parameter look up)
+    container_type = type(torsion)
+    assert len(set(torsion)) == len(torsion)
+    _a, b, _c, _d = torsion
+    a, c, d = sorted([_a, _c, _d])
+    canonicalized_1 = container_type([a, b, c, d])
+
+    # approach 2: take min((b, a, c, d), (d, c, a, b)) (used for applying trefoil convention)
+    a, b, c, d = torsion
+    canonicalized_2 = canonicalize_bonded_ixn([b, a, c, d])
+
+    # TODO: which one?
+    # return canonicalized
+    raise NotImplementedError(f"can't decide between {canonicalized_1} and {canonicalized_2}!")
```

Comment:
I think it makes sense to go with approach 1 here, and possibly add a separate function to enumerate the trefoil permutations

Reply:
Have now split this into a couple functions, and added documentation on points that tripped me up